src/main/kotlin/org/jetbrains/research/ml/codetracker/tracking/ActivityTrackerFileHandler.kt

```diff
@@ -0,0 +1,73 @@
+package org.jetbrains.research.ml.codetracker.tracking
+
+import com.intellij.openapi.diagnostic.Logger
+import krangl.*
+import org.apache.commons.csv.CSVFormat
+import org.jetbrains.research.ml.codetracker.Plugin.PLUGIN_ID
+import org.jetbrains.research.ml.codetracker.models.Language
+import org.jetbrains.research.ml.codetracker.server.PluginServer
+import java.io.File
+import java.io.FileNotFoundException
+
+enum class ActivityTrackerColumn {
+    TIMESTAMP, USERNAME, EVENT_TYPE, EVENT_DATA, PROJECT_NAME, FOCUSED_COMPONENT,
+    CURRENT_FILE, PSI_PATH, EDITOR_LINE, EDITOR_COLUMN
+}
+
+object ActivityTrackerFileHandler {
+
+    private const val ACTIVITY_TRACKER_FILE_NAME = "ide-events"
+    private const val DEFAULT_PATH_SYMBOL = "*"
+    private val logger: Logger = Logger.getInstance(javaClass)
+
+    // TODO: get the current language instead of the argument??
+    fun filterActivityTrackerData(filePath: String, language: Language = Language.PYTHON): String? {
+        return try {
+            val df = DataFrame.readCSV(
+                filePath,
+                format = CSVFormat.DEFAULT.withHeader(ActivityTrackerColumn::class.java)
+            )
+            val filteredDf = filterDataFrame(df, language)
+            val resultPath = filePath.replace(ACTIVITY_TRACKER_FILE_NAME, "${ACTIVITY_TRACKER_FILE_NAME}_filtered")
+            filteredDf.writeCSV(File(resultPath), format = CSVFormat.DEFAULT.withIgnoreHeaderCase(true))
+            resultPath
+        } catch (e: FileNotFoundException) {
+            logger.info("${PLUGIN_ID}: The activity tracker file $filePath does not exist")
+            null
+        }
+    }
+
+    private fun filterDataFrame(df: DataFrame, language: Language): DataFrame {
+        // Remove columns, which can contain private information
+        val anonymousDf = df.remove(
+            ActivityTrackerColumn.USERNAME.name, ActivityTrackerColumn.PROJECT_NAME.name,
+            ActivityTrackerColumn.PSI_PATH.name
+        )
+        return clearFilesPaths(anonymousDf, language)
+    }
+
+    // Return default symbol, if the file
+    private fun replaceAbsoluteFilePath(path: String, pluginFilesRegex: Regex): String {
+        // Try to find plugin's tasks files
+        val found = pluginFilesRegex.find(path)
+        return if (found != null) {
+            // Get name from path
+            path.split("/").last()
+        } else {
+            DEFAULT_PATH_SYMBOL
+        }
+    }
+
+    private fun clearFilesPaths(df: DataFrame, language: Language): DataFrame {
+        val tasks = PluginServer.tasks.joinToString(separator = "|") { it.key }
+        val tasksMatchCondition = ".*/$PLUGIN_ID/($tasks)${language.extension.ext}".toRegex(
+            RegexOption.IGNORE_CASE
+        )
+        // Replace the absolute file path into something like this: PLUGIN_ID/task_file
```

Comment:
into -> **with** or **by**

Also, shouldn't we mention that in case of non-plugin files the replacement will be DEFAULT_PATH_SYMBOL?

Reply:
I think we should delete the comment because the `replaceAbsoluteFilePath` function's description contains this information